distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1145,43 +1155,69 @@ public void testDownloadFromFederatedCswSource() throws Exception {
      */
     @Test
     public void testRetrievalReliablility() throws Exception {
-        cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL, ACTIVITIES_CHANNEL));
+        getSecurityPolicy().configureWebContextPolicy(null,
+                "/=SAML|basic,/solr=SAML|PKI|basic",
+                null,
+                null);
+        localhostCometDClient = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
 
         String filename = "product2.txt";
         String metacardId = generateUniqueMetacardId();
         String resourceData = getResourceData(metacardId);
 
-        cswServer.whenHttp()
-                .match(post("/services/csw"),
-                        withPostBodyContaining("GetRecords"),
-                        withPostBodyContaining(metacardId))
-                .then(ok(),
-                        contentType("text/xml"),
-                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
-
-        cswServer.whenHttp()
-                .match(Condition.get("/services/csw"),
-                        Condition.parameter("request", "GetRecordById"),
-                        Condition.parameter("id", metacardId))
-                .then(getCswRetrievalHeaders(filename),
-                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 2));
+        setupStubCswResponse(filename, metacardId, resourceData);
 
         String restUrl = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + metacardId
-                + "?transform=resource" + "&session=" + cometDClient.getClientId();
+                + "?transform=resource";
```

Comment:
Possibly cleaner: 

```
String restUrl = String.format("%ssources/%s/%s?transform=resource", 
         REST_PATH.getUrl(),
         CSW_STUB_SOURCE_ID,
         metacardId);
```

Although that "%ssources" could be confusing ... so if we wanted to pull out `REST_PATH.getUrl()` I could see the benefit: 

```
String restUrl = REST_PATH.getUrl() + String.format("...", ... );
```